app/response_processor.py

```diff
@@ -0,0 +1,175 @@
+import logging
+import settings
+from settings import session
+import zipfile
+import io
+import pika
+from structlog import wrap_logger
+from ftplib import FTP
+from requests.packages.urllib3.exceptions import MaxRetryError
+
+logging.basicConfig(level=settings.LOGGING_LEVEL, format=settings.LOGGING_FORMAT)
+
+logger = logging.getLogger(__name__)
+
+logger = wrap_logger(logger)
+logger.debug("START")
+
+
+def connect_to_ftp():
+    ftp = FTP(settings.FTP_HOST)
+    ftp.login(user=settings.FTP_USER, passwd=settings.FTP_PASS)
+    return ftp
+
+
+def deliver_ascii_to_ftp(ftp, filename, data):
+    stream = io.StringIO(data)
+    ftp.storlines('STOR ' + filename, stream)
+
+
+def deliver_binary_to_ftp(ftp, filename, data):
+    stream = io.BytesIO(data)
+    ftp.storbinary('STOR ' + filename, stream)
+
+
+def remote_call(request_url, json=None):
+    try:
+        r = None
+
+        if json:
+            r = session.post(request_url, json=json)
+        else:
+            r = session.get(request_url)
+
+        return r
+    except MaxRetryError:
+        logger.error("Max retries exceeded (5)", request_url=request_url)
+
+
+def response_ok(res, error_on_fail, **kwargs):
+    if res.status_code != 200:
+        logger.error(error_on_fail, request_url=res.url, **kwargs)
+        return False
+    return True
+
+
+class ResponseProcessor:
+    def __init__(self, logger):
+        self.logger = logger
+
+    def process(self, mongoid):
+        processed_ok = False
+        survey_response = self.get_doc_from_store(mongoid)
+
+        if survey_response:
+            metadata = survey_response['metadata']
+
+            # Update consumers logger to use bound vars
+            self.logger = logger.bind(user_id=metadata['user_id'], ru_ref=metadata['ru_ref'])
+            sequence_no = self.get_sequence_no()
+
+        if survey_response and sequence_no:
+            if 'file-type' in survey_response and survey_response['file-type'] == 'xml':
+                xml_content = self.transform_xml(survey_response)
+
+                if xml_content:
+                    processed_ok = self.notify_queue(xml_content)
+
+                return processed_ok
+            else:
+                zip_contents = self.transform_cs(sequence_no, survey_response)
+
+                if zip_contents:
+                    processed_ok = self.process_zip(zip_contents)
+
+                return processed_ok
+
+    def get_doc_from_store(self, mongoid):
+        """Retrieve a doc from the store. Bind a logger to a user/ru_ref
+
+        :rtype boolean: Whether the doc was retrieved successfully
+
+        """
+        store_url = settings.SDX_STORE_URL + "/responses/" + mongoid
+
+        r = remote_call(store_url)
+
+        if not response_ok(r, "Store retrieval failed", document_id=mongoid):
```

Comment:
Doesn't feel right to have an affirmative method that we always call with 'if not', also feels wrong to pass an error message that may or may not happen. How about:
```
if r.status_code != 200:
    log_request_not_ok(r, "Store retrieval failed", document_id=mongoid)
    return False
```